independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -0,0 +1,260 @@
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_NAME;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_OS;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_OS_ARCH;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_QUARKUS;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_TIMEZONE;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_VENDOR;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.PROP_VERSION;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.VALUE_NULL_IP;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.CommonSystemProperties.GRAALVM_VERSION_DISTRIBUTION;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.CommonSystemProperties.GRAALVM_VERSION_JAVA;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.CommonSystemProperties.GRAALVM_VERSION_VERSION;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.CommonSystemProperties.GRADLE_VERSION;
+import static io.quarkus.analytics.dto.segment.ContextBuilder.CommonSystemProperties.MAVEN_VERSION;
+import static io.quarkus.analytics.rest.RestClient.TTL;
+import static io.quarkus.analytics.util.StringUtils.hashSHA256;
+import static io.quarkus.maven.dependency.DependencyFlags.TOP_LEVEL_RUNTIME_EXTENSION_ARTIFACT;
+import static java.lang.System.getProperty;
+import static java.lang.System.getenv;
+import static java.util.Optional.ofNullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.TextStyle;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.quarkus.analytics.config.FileLocations;
+import io.quarkus.analytics.dto.config.Identity;
+import io.quarkus.analytics.dto.segment.ContextBuilder;
+import io.quarkus.analytics.dto.segment.Track;
+import io.quarkus.analytics.dto.segment.TrackEventType;
+import io.quarkus.analytics.dto.segment.TrackProperties;
+import io.quarkus.analytics.rest.RestClient;
+import io.quarkus.analytics.util.FileUtils;
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.bootstrap.workspace.WorkspaceModuleId;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+import io.quarkus.maven.dependency.ArtifactCoords;
+
+public class AnalyticsService {
+    private static AnalyticsService INSTANCE = null;
+    private final Queue<CompletableFuture<HttpResponse<String>>> postFutures;
+
+    public static AnalyticsService getInstance(final FileLocations fileLocations, MessageWriter log) {
+        if (INSTANCE == null) {
```

Comment:
Especially in Gradle, it'll likely cause issues having a singleton of this class that holds build/task state (e.g. the `MessageWriter` instance is derived from the first executed Gradle task. Not sure how `mvnd` behaves (whether it keeps plugin classes around).

Reply:
will move away from singletons.

Reply:
done